pkg/controller/cronjob/cronjob_controllerv2_test.go

```diff
@@ -186,74 +186,77 @@ func TestControllerV2SyncCronJob(t *testing.T) {
-		"prev ran but done, not time, F":                {f, F, onTheHour, noDead, T, F, justAfterThePriorHour(), justBeforeTheHour(), F, F, 0, 0, F, T, F, T, nil},
-		"prev ran but done, not time, R":                {R, F, onTheHour, noDead, T, F, justAfterThePriorHour(), justBeforeTheHour(), F, F, 0, 0, F, T, F, T, nil},
-		"prev ran but done, is time, A":                 {A, F, onTheHour, noDead, T, F, justAfterThePriorHour(), *justAfterTheHour(), T, F, 1, 0, F, T, F, T, nil},
-		"prev ran but done, is time, F":                 {f, F, onTheHour, noDead, T, F, justAfterThePriorHour(), *justAfterTheHour(), T, F, 1, 0, F, T, F, T, nil},
-		"prev ran but done, is time, R":                 {R, F, onTheHour, noDead, T, F, justAfterThePriorHour(), *justAfterTheHour(), T, F, 1, 0, F, T, F, T, nil},
-		"prev ran but done, is time, suspended":         {A, T, onTheHour, noDead, T, F, justAfterThePriorHour(), *justAfterTheHour(), F, F, 0, 0, F, F, F, T, nil},
-		"prev ran but done, is time, past deadline":     {A, F, onTheHour, shortDead, T, F, justAfterThePriorHour(), *justAfterTheHour(), F, F, 0, 0, F, T, F, T, nil},
-		"prev ran but done, is time, not past deadline": {A, F, onTheHour, longDead, T, F, justAfterThePriorHour(), *justAfterTheHour(), T, F, 1, 0, F, T, F, T, nil},
-
-		"still active, not time, A":                {A, F, onTheHour, noDead, T, T, justAfterThePriorHour(), justBeforeTheHour(), F, F, 1, 0, F, T, F, T, nil},
-		"still active, not time, F":                {f, F, onTheHour, noDead, T, T, justAfterThePriorHour(), justBeforeTheHour(), F, F, 1, 0, F, T, F, T, nil},
-		"still active, not time, R":                {R, F, onTheHour, noDead, T, T, justAfterThePriorHour(), justBeforeTheHour(), F, F, 1, 0, F, T, F, T, nil},
-		"still active, is time, A":                 {A, F, onTheHour, noDead, T, T, justAfterThePriorHour(), *justAfterTheHour(), T, F, 2, 0, F, T, F, T, nil},
-		"still active, is time, F":                 {f, F, onTheHour, noDead, T, T, justAfterThePriorHour(), *justAfterTheHour(), F, F, 1, 0, F, T, F, T, nil},
-		"still active, is time, R":                 {R, F, onTheHour, noDead, T, T, justAfterThePriorHour(), *justAfterTheHour(), T, T, 1, 0, F, T, F, T, nil},
-		"still active, is time, suspended":         {A, T, onTheHour, noDead, T, T, justAfterThePriorHour(), *justAfterTheHour(), F, F, 1, 0, F, F, F, T, nil},
-		"still active, is time, past deadline":     {A, F, onTheHour, shortDead, T, T, justAfterThePriorHour(), *justAfterTheHour(), F, F, 1, 0, F, T, F, T, nil},
-		"still active, is time, not past deadline": {A, F, onTheHour, longDead, T, T, justAfterThePriorHour(), *justAfterTheHour(), T, F, 2, 0, F, T, F, T, nil},
+		"never ran, not valid schedule, A":      {A, F, errorSchedule, noDead, F, F, justAfterThePriorHour(), justBeforeTheHour(), F, F, 0, 1, F, F, F, F, T, nil, nil},
```

Comment:
Gulp...

Can you open a separate PR to spell out the structs?

```golang
{
  concurrencyPolicy: A,
  schedule:              errorSchedule,
 ....
}
```

We shouldn't be listing the properties that don't matter (like `nil`, `false`). Also please get rid of constants `F` and `T`.

Reply:
It's very hard to read were you added the test, but I'm gonna trust you that you added an appropriate test for now.

Reply:
> 

Sure. I will raise a issue and submit a  PR for that

Reply:
forgive me. I should mention, the two cases are "prev ran but done, is time, create job failed, A" and "still active, is time, get job failed, R"

Reply:
https://github.com/kubernetes/kubernetes/issues/107633 has been raised to spell out the structs

Reply:
Since we have time before code freeze, can you fix #107633 first?

Reply:
Yeah, I am working on #107633, hopefully the PR will be submitted tomorrow.

Reply:
Yeah, I am working on #107633 , the PR will be submitted tomorrow.

I am not sure, could we pick up the current PR first. Since I am working on #107633 based on the change of 107470. This sequence can save me some time on cherry pick or conflict fixing.

Reply:
The problem is that is is pretty hard to review the tests as it is.

The code LGTM. @soltysh WDYT?

Reply:
Sorry for the tough task, I did not realize that from reviewer perspective.

i will fix #107633 first, and then rebase #107470.

Reply:
PR https://github.com/kubernetes/kubernetes/pull/107659 for #107633 has been submitted